flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherRestEndpoint.java

```diff
@@ -77,8 +74,11 @@ protected void setupChannelHandlers(Router router, CompletableFuture<String> res
 			optWebContent = Optional.empty();
 		}
 
-		optWebContent.ifPresent(
-			webContentHandler -> router.GET("/:*", webContentHandler));
+		return optWebContent
+			.map(webContent ->
+				Collections.singleton(
+					Tuple2.<RestHandlerSpecification, ChannelInboundHandler>of(WebContentHandlerSpecification.getInstance(), webContent)))
+			.orElseGet(() -> Collections.emptySet());
```

Comment:
Given that we're adding new handlers soon we may want to write things so that we can more easily extend the collection it without rewriting most of this block.

```
Collection<Tuple2<RestHandlerSpecification, ChannelInboundHandler>> result = new ArrayList<>();
optWebContent
	.map(webContent -> result.add(webContent));
return result;
```

Reply:
I would like to do this in one of the follow up PRs where I actually add more REST handlers, if that's ok.

Reply:
sure